http/src/config.rs

```diff
@@ -146,10 +130,28 @@ fn create(
         },
     };
 
-    serde_json::to_writer_pretty(BufWriter::new(config), &config_contents)
+    let config_path = ipfs_path.join("config");
+
+    let config_file = fs::create_dir_all(&ipfs_path)
+        .map_err(InitializationError::DirectoryCreationFailed)
+        .and_then(|_| {
+            fs::File::create(&config_path).map_err(InitializationError::ConfigCreationFailed)
+        })?;
+
+    serde_json::to_writer_pretty(BufWriter::new(config_file), &config_contents)
         .map_err(InitializationError::ConfigWritingFailed)?;
 
```

Comment:
Probably an older bug: since we are using `BufWriter` and `serde_json::to_writer_pretty` does not explicitly say it'll flush the writer (std::io::Write::flush), perhaps we should pass the writer as `&mut` and take care of this ourselves? This will otherwise be done in `drop` and the error would be lost.